djstripe/models.py

```diff
@@ -2753,6 +2753,27 @@ class Product(StripeObject):
     ))
     unit_label = StripeCharField(max_length=12, null=True)
 
+    @classmethod
+    def get_or_create(cls, **kwargs):
+        """ Get or create a Product."""
+
+        try:
+            return Product.objects.get(stripe_id=kwargs['stripe_id']), False
+        except Product.DoesNotExist:
+            return cls.create(**kwargs), True
+
+    @classmethod
+    def create(cls, **kwargs):
+        # A few minor things are changed in the api-version of the create call
+        api_kwargs = dict(kwargs)
+        api_kwargs['id'] = api_kwargs['stripe_id']
+        del (api_kwargs['stripe_id'])
```

Comment:
This stuff doesn't look right, it applies to every single stripe object, which means it should already be handled.

Reply:
I just copied it from Plan.create() https://github.com/dj-stripe/dj-stripe/blob/cc64c084f2f5c92d9aca60cf81c064bfa1fcde7f/djstripe/models.py#L2636

Reply:
Yikes. I guess we're in worse shape than I thought.

Reply:
Hah. I had also wondered what the purpose of Class.get_or_create() was, is there a reason you don't call _api_create on creation of the new model and then just use Class.objects.get_or_create() instead?

Reply:
I don't really know, a lot of it predates my maintainership. I'm not a fan of those methods that are straight on the class, but my approach has been to leave them alone and neither add nor remove until I have a plan for them.

Such a plan might be an `api_objects` manager for example. Dunno, hard to say.

Reply:
That was my thinking too, though I thought in this case as it was a clear copy of functionality when a solution is found both chunks of code could be fixed together.

Reply:
> Such a plan might be an api_objects manager for example. Dunno, hard to say.

I'm a fan of this plan.